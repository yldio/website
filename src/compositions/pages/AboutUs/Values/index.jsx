import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import TileGrid, { Tile } from 'components/TileGrid';
import PageSection from 'components/PageSection';
import { Copy, H2 } from 'components/Typography';
import {
  TileContent,
  TileContentTitle,
  TileContentBody,
  Header,
  Description,
  OurValues,
  ValuesSection,
} from './styled';

export const Value = ({ header, children }) => (
  <ValuesSection>
    <Header>{header}</Header>
    <Description>{children}</Description>
  </ValuesSection>
);

const Values = () => (
  <Fragment>
    <PageSection data-selector="about-us:our-values">
      <H2>Our Values</H2>

      <Copy light>
        We help you future proof your business though a new style of consulting,
        ground up. We connect and educate engineering, product and leadership to
        help you navigate digital transformation, grow your business and tackle
        core customer problems. We’re one of the most renowned technology
        companies in Europe.
      </Copy>
      <Copy>
        Some amazing support copy with go here to reenforce whatever section it
        is relevant too. Some amazing support copy with go here to reenforce
        whatever section it is relevant too. Some amazing support copy with go
        here to reenforce whatever section it is relevant too. Some amazing
        support copy with go here to reenforce whatever section it is relevant
        too.
      </Copy>
    </PageSection>
    <PageSection>
      <TileGrid>
        <OurValues>
          <Value header="Your opportunity is limitless">
            You are the driver of change. We never give up, we begin again. Be
            adventuruous. Push the boundaries. Go the extra mile.
          </Value>
          <Value header="Have integrity, be honest">
            Lead by example. No “bullshit”. Be real, have respect.
          </Value>
          <Value header="Get better as a team">
            Collaborate, collaborate, collaborate. Never stop learning annd
            learn together - innovation can be the innovation you inspire in
            others. Be inquisitive. Ask why, and be preparered to be asked why
            of yourself.
          </Value>
          <Value header="Recognise success">
            Understand the purpose. When you succeed, help others do the same.
            Celebrate, even when we fail.
          </Value>
          <Value header="Have courage, stay humble">
            Put our partners success ahead of our own and treat them as we would
            our own team. Have strong opinions, weakly held. Good enough isn’t
            good enough.
          </Value>
        </OurValues>
      </TileGrid>
    </PageSection>
    <PageSection wide>
      <TileGrid>
        <Tile>
          <TileContent>
            <TileContentTitle>Services</TileContentTitle>
            <TileContentBody>
              We enable your organisation to deliver world-class technology and
              user experiences, long after we leave.
            </TileContentBody>
          </TileContent>
        </Tile>
        <Tile colour="darker">
          <TileContent>
            <TileContentTitle>Community</TileContentTitle>
            <TileContentBody>
              We define ourselves by the people who represent us.
            </TileContentBody>
          </TileContent>
        </Tile>
      </TileGrid>
    </PageSection>
  </Fragment>
);

export default Values;

Value.propTypes = {
  header: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
};
