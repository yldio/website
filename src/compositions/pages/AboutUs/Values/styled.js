import styled from 'styled-components';
import { spacing } from 'styles/spacing';
import { textOnDark, highlight } from 'styles/colours';

export const OurValues = styled.div`
  background-color: ${highlight};
  color: ${textOnDark};
  padding: 69px 124px 69px 65px;
`;

export const TileContent = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: ${spacing(4)};
`;

export const TileContentTitle = styled.h5`
  letter-spacing: 1px;
  font-size: 1.125rem;
  font-weight: 700;
  text-transform: uppercase;
`;

export const TileContentBody = styled.p`
  font-weight: 300;
  font-size: 1.5rem;
`;

export const Tile = styled.div`
  display: inline-block;
  width: 100%;
`;
export const Header = styled.div`
  display: inline-block;
  width: 50%;
  font-weight: 300;
  font-size: 1.5rem;
`;
export const Description = styled.div`
  display: inline-block;
  width: 50%;
  font-weight: normal;
  font-size: 1.125rem;
`;

export const ValuesSection = styled.div`
  width: 100%;
  min-height: 10.63rem;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
`;
