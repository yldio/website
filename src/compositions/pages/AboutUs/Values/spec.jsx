import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Values, { Value } from './index';

test('components/Values snapshot', () => {
  const wrapper = shallow(<Values />);

  expect(wrapper).toMatchSnapshot();
});

test('components/Value snapshot', () => {
  const wrapper = shallow(
    <Value header="Your opportunity is limitless">
      You are the driver of change. We never give up, we begin again. Be
      adventuruous. Push the boundaries. Go the extra mile.
    </Value>,
  );

  expect(wrapper.find('styled__Header').html()).toBe(
    '<div class="styled__Header-ctRlcN fOoAaV">Your opportunity is limitless</div>',
  );

  expect(wrapper).toMatchSnapshot();
});
