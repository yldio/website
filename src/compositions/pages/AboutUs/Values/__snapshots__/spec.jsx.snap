// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Value snapshot 1`] = `
<styled__ValuesSection>
  <styled__Header>
    Your opportunity is limitless
  </styled__Header>
  <styled__Description>
    You are the driver of change. We never give up, we begin again. Be adventuruous. Push the boundaries. Go the extra mile.
  </styled__Description>
</styled__ValuesSection>
`;

exports[`components/Values snapshot 1`] = `
<React.Fragment>
  <PageSection
    data-selector="about-us:our-values"
    gradient={false}
    gradientHeight="50%"
  >
    <Typography__H2>
      Our Values
    </Typography__H2>
    <Typography__base
      light={true}
    >
      We help you future proof your business though a new style of consulting, ground up. We connect and educate engineering, product and leadership to help you navigate digital transformation, grow your business and tackle core customer problems. We’re one of the most renowned technology companies in Europe.
    </Typography__base>
    <Typography__base>
      Some amazing support copy with go here to reenforce whatever section it is relevant too. Some amazing support copy with go here to reenforce whatever section it is relevant too. Some amazing support copy with go here to reenforce whatever section it is relevant too. Some amazing support copy with go here to reenforce whatever section it is relevant too.
    </Typography__base>
  </PageSection>
  <PageSection
    gradient={false}
    gradientHeight="50%"
  >
    <TileGrid>
      <styled__OurValues>
        <Value
          header="Your opportunity is limitless"
        >
          You are the driver of change. We never give up, we begin again. Be adventuruous. Push the boundaries. Go the extra mile.
        </Value>
        <Value
          header="Have integrity, be honest"
        >
          Lead by example. No “bullshit”. Be real, have respect.
        </Value>
        <Value
          header="Get better as a team"
        >
          Collaborate, collaborate, collaborate. Never stop learning annd learn together - innovation can be the innovation you inspire in others. Be inquisitive. Ask why, and be preparered to be asked why of yourself.
        </Value>
        <Value
          header="Recognise success"
        >
          Understand the purpose. When you succeed, help others do the same. Celebrate, even when we fail.
        </Value>
        <Value
          header="Have courage, stay humble"
        >
          Put our partners success ahead of our own and treat them as we would our own team. Have strong opinions, weakly held. Good enough isn’t good enough.
        </Value>
      </styled__OurValues>
    </TileGrid>
  </PageSection>
  <PageSection
    gradient={false}
    gradientHeight="50%"
    wide={true}
  >
    <TileGrid>
      <styled__Tile>
        <styled__TileContent>
          <styled__TileContentTitle>
            Services
          </styled__TileContentTitle>
          <styled__TileContentBody>
            We enable your organisation to deliver world-class technology and user experiences, long after we leave.
          </styled__TileContentBody>
        </styled__TileContent>
      </styled__Tile>
      <styled__Tile
        colour="darker"
      >
        <styled__TileContent>
          <styled__TileContentTitle>
            Community
          </styled__TileContentTitle>
          <styled__TileContentBody>
            We define ourselves by the people who represent us.
          </styled__TileContentBody>
        </styled__TileContent>
      </styled__Tile>
    </TileGrid>
  </PageSection>
</React.Fragment>
`;
